src/coreclr/src/vm/method.cpp

```diff
@@ -5342,29 +5342,21 @@ LPVOID NDirectMethodDesc::FindEntryPoint(NATIVE_LIBRARY_HANDLE hMod) const
         return reinterpret_cast<LPVOID>(pFunc);
     }
 
-    DWORD probedEntrypointNameLength = (DWORD)(strlen(funcName) + 1); // +1 for charset decorations
-
     // Allocate space for a copy of the entry point name.
+    DWORD probedEntrypointNameLength = (DWORD)(strlen(funcName) + 1); // +1 for charset decorations
     int dstbufsize = (int)(sizeof(char) * (probedEntrypointNameLength + 1)); // +1 for the null terminator
-
     LPSTR szProbedEntrypointName = ((LPSTR)_alloca(dstbufsize));
 
     // Copy the name so we can mangle it.
     strcpy_s(szProbedEntrypointName, dstbufsize, funcName);
-    szProbedEntrypointName[probedEntrypointNameLength] = '\0'; // Add an extra '\0'.
+    szProbedEntrypointName[probedEntrypointNameLength] = '\0'; // Null terminator
+    szProbedEntrypointName[probedEntrypointNameLength - 1] = IsNativeAnsi() ? 'A' : 'W'; // Charset suffix
 
-    if(!IsNativeNoMangled())
```

Comment:
Why is it ok to remove this check?

Reply:
We should already have returned due to line 5340 above if `IsNativeNoMangled()` is true.

Reply:
Ah, GitHub was hiding that in the diff. Thanks.